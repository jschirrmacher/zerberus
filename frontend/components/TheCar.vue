<script setup lang="ts">
import CarWheel from "./CarWheel.vue"
import CarChassis from "./CarChassis.vue"
</script>

<template>
  <div class="car">
    <!-- <CarWheel /> -->
    <!-- <CarWheel /> -->
    <CarChassis />
    <!-- <CarWheel />
    <CarWheel /> -->
  </div>
</template>

<style scoped lang="scss"></style>
