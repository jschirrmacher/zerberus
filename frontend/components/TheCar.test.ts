import { describe, it, expect } from "vitest"
import { shallowMount } from "@vue/test-utils"
import TheCar from "./TheCar.vue"

describe("TheCar", () => {
  it("should render as expected", () => {
    const component = shallowMount(TheCar, {
      props: { motorSpeedLeft: 0, motorSpeedRight: 0, direction: 0 },
    })
    expect(component.element).toMatchSnapshot()
  })
})
