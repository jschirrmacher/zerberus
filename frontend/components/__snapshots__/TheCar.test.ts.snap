// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheCar > should render as expected 1`] = `
<div
  class="car"
  data-v-fb2245a3=""
>
  <car-chassis-stub
    color="#000000"
    data-v-fb2245a3=""
    height="125"
    length="250"
    width="500"
  />
  <div
    class="left"
    data-v-fb2245a3=""
  >
    <div
      class="back"
      data-v-fb2245a3=""
    >
      <car-wheel-stub
        data-v-fb2245a3=""
        diameter="180"
        speed="0"
        width="50"
      />
    </div>
    <div
      class="front"
      data-v-fb2245a3=""
    >
      <car-wheel-stub
        data-v-fb2245a3=""
        diameter="180"
        speed="0"
        width="50"
      />
    </div>
  </div>
  <div
    class="right"
    data-v-fb2245a3=""
  >
    <div
      class="back"
      data-v-fb2245a3=""
    >
      <car-wheel-stub
        data-v-fb2245a3=""
        diameter="180"
        speed="0"
        width="50"
      />
    </div>
    <div
      class="front"
      data-v-fb2245a3=""
    >
      <car-wheel-stub
        data-v-fb2245a3=""
        diameter="180"
        speed="0"
        width="50"
      />
    </div>
  </div>
</div>
`;
